src/batch_iterator.cpp

```diff
@@ -112,47 +112,35 @@ fixed_buffer_map* batch_iterator_fbm::filler()
     fixed_buffer_map* rc = get_pending_buffer();
     m_state              = async_state::processing;
 
-    // This is for the first pass
     if (m_input_ptr == nullptr)
     {
-        m_state     = async_state::fetching_data;
         m_input_ptr = m_source->next();
         m_src_index = 0;
-        m_state     = async_state::processing;
     }
 
-    m_dst_index      = 0;
-    size_t remainder = m_batch_size;
-    while (remainder > 0)
-    {
-        if (m_input_ptr == nullptr)
-        {
-            rc = nullptr;
-            break;
-        }
-
-        const string first_name         = (m_input_ptr->get_names())[0];
-        size_t       input_size         = ((*m_input_ptr)[first_name])->get_item_count();
-        size_t       current_input_size = input_size - m_src_index;
-        size_t move_count = (current_input_size <= remainder) ? current_input_size : remainder;
+    if (m_input_ptr == nullptr)
```

Comment:
What do you think about moving this if statement inside the if statement above?

Reply:
ok